src/python/pants/backend/jvm/targets/jvm_target.py

```diff
@@ -104,16 +104,22 @@ def __init__(self,
       entity_description='fatal_warnings',
       hint_message="fatal_warnings should be defined as part of the target compiler_option_sets"
     )
-    if fatal_warnings and 'fatal_warnings' not in compiler_option_sets:
-      compiler_option_sets.append('fatal_warnings')
+    if fatal_warnings is not None:
```

Comment:
This bunch of code is handling the case where `fatal_warnings` is set to false on the target, but it might be also either in the `compiler_option_sets` target params or CLI options. 
This is the minimal way I could think of, thoughts?
I'd like to see this all removed once `fatal_warnings` is removed as an actual option/param :P

Reply:
One option might be to have the internal representation of `compiler_options_sets` temporarily be a dict from key to book?

Reply:
...oh. This is already nicer than that.